appcues/src/main/java/com/appcues/ui/utils/AppcuesAspectRatioModifier.kt

```diff
@@ -78,88 +94,72 @@ internal class AppcuesAspectRatioModifier(
-            tryMaxHeight(enforceConstraints = false).also { if (it != IntSize.Zero) return it }
-            tryMaxWidth(enforceConstraints = false).also { if (it != IntSize.Zero) return it }
-            tryMinHeight(enforceConstraints = false).also { if (it != IntSize.Zero) return it }
-            tryMinWidth(enforceConstraints = false).also { if (it != IntSize.Zero) return it }
+            tryMatchWidthFirst(aspectRatio, widthPixels, true) ?: tryMatchWidthFirst(aspectRatio, widthPixels, false)
         }
-        return IntSize.Zero
+
+        return size ?: IntSize.Zero
     }
 
-    private fun Constraints.tryMaxWidth(enforceConstraints: Boolean = true): IntSize {
-        val maxWidth = this.maxWidth
-        if (maxWidth != Constraints.Infinity) {
-            val height = (maxWidth / aspectRatio).roundToInt()
-            if (height > 0) {
-                val size = IntSize(maxWidth, height)
-                if (!enforceConstraints || isSatisfiedBy(size)) {
-                    return size
-                }
-            }
-        }
-        return IntSize.Zero
+    private fun Constraints.shouldMatchHeightFirst(
+        containerAspectRatio: Float,
+        widthPixels: Float?,
+        parentAspectRatio: Float?
+    ) = when (contentMode) {
+        FILL -> shouldMatchHeightFirstFill(widthPixels, containerAspectRatio)
+        FIT -> shouldMatchHeightFirstFit(containerAspectRatio, parentAspectRatio, widthPixels)
     }
 
-    private fun Constraints.tryMaxHeight(enforceConstraints: Boolean = true): IntSize {
-        val maxHeight = this.maxHeight
-        if (maxHeight != Constraints.Infinity) {
-            val width = (maxHeight * aspectRatio).roundToInt()
-            if (width > 0) {
-                val size = IntSize(width, maxHeight)
-                if (!enforceConstraints || isSatisfiedBy(size)) {
-                    return size
+    private fun Constraints.shouldMatchHeightFirstFill(
+        widthPixels: Float?,
+        containerAspectRatio: Float
+    ): Boolean {
+        when (stackScope) {
+            is ColumnStackScope -> {}
+            is RowStackScope -> {
+                if (stackScope.greaterHeight.value > 0) {
+                    aspectRatio = if (widthPixels != null) {
+                        widthPixels.toFloat() / stackScope.greaterHeight.value
+                    } else {
+                        this.maxWidth.toFloat() / stackScope.greaterHeight.value
+                    }
                 }
             }
         }
-        return IntSize.Zero
+
+        return containerAspectRatio < aspectRatio
     }
 
-    private fun Constraints.tryMinWidth(enforceConstraints: Boolean = true): IntSize {
-        val minWidth = this.minWidth
-        val height = (minWidth / aspectRatio).roundToInt()
-        if (height > 0) {
-            val size = IntSize(minWidth, height)
-            if (!enforceConstraints || isSatisfiedBy(size)) {
-                return size
-            }
+    @Suppress("unused")
```

Comment:
is this an unused private fun?

Reply:
its just the Constraint. that is unused

Reply:
oh, so should the extension function be on something else? granted, not a big deal here it seems.